server/controllers/nlp.js

```diff
@@ -17,19 +21,176 @@ let functions = {
                 type: 'histogram'
             }]
         ))
-    }
+    },
+
+    plotLineChartOfColumn: function (dataset, parameters, callback) {
+        let column = parameters[0];
+
+        // SELECT <column> FROM <dataset>
+        bookshelf.Model.extend({tableName: dataset}).fetchAll({columns: [column]}).then(data => callback(
+            [{
+                // map [{<columnName>: <columnValue>}, ...] to [<columnValue>, ...]
+                x: data.map((value) => value.attributes[column]),
+                type: 'scatter'
+            }]
+        ))
+    },
+    plotScatterOfTwoColumns: function (dataset, parameters, callback) {
+        let column1 = parameters[0];
+        let column2 = parameters[1];
+
+        // SELECT <column> FROM <dataset>
+        bookshelf.Model.extend({tableName: dataset}).fetchAll({columns: [column1, column2]}).then(data => {
+            callback(
+                [{
+                    // map [{<columnName>: <columnValue>}, ...] to [<columnValue>, ...]
+                    x: data.map((value) => value.attributes[column1]),
+                    y: data.map((value) => value.attributes[column2]),
+                    mode: 'markers',
+                    type: 'scatter'
+                }]
+            )
+        });
+    },
+    plotPieChartOfColumn: function (dataset, parameters, callback) {
+        let column = parameters[0];
+        // SELECT <column> FROM <dataset>
```

Comment:
this should be updated as well

Reply:
what should be updated?

Reply:
you are doing a groupting and counting here, right? can we do this directly on the db? if yes, please update the sql "query" as well, if not, how can we document the data transformations that happen here?

Reply:
a query would look like this `SELECT count(*), <column> FROM <dataset> GROUP BY <column>`

Reply:
did you do some research on this?

Reply:
Nope i think this should be done in the next iteration

Reply:
ok, could you then create a ticket for this please?